segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/IndexReader.java

```diff
@@ -104,6 +104,26 @@ long getBucketCount(SegmentProperties segmentInfo) {
         return segmentInfo.getAttributes().getOrDefault(TableAttributes.BUCKET_COUNT, 0L);
     }
 
+    /**
+     * Gets the index in the Segment until which compaction has previously run.
+     *
+     * @param segmentInfo A {@link SegmentProperties} from which to extract the information.
+     * @return The index.
+     */
+    long getCompactionOffset(SegmentProperties segmentInfo) {
+        return segmentInfo.getAttributes().getOrDefault(TableAttributes.COMPACTION_OFFSET, 0L);
+    }
+
+    /**
+     * Gets the Segment Utilization (as a percentage) under which a compaction may be required.
+     *
+     * @param segmentInfo A {@link SegmentProperties} from which to extract the information.
+     * @return The result.
+     */
+    long getCompactionUtilizationThreshold(SegmentProperties segmentInfo) {
```

Comment:
The fact that this is returned as a long makes the calculations more complicated. 
We can have the value in the config as a percentage and here divide by 100.0 and return the double.